presto-accumulo/src/main/java/com/facebook/presto/accumulo/conf/AccumuloConfig.java

```diff
@@ -0,0 +1,333 @@
+        }
+    }
+
+    /**
+     * Gets the class name of the configured metadata manager. Default is the class name of the
+     * class from
+     * {@link AccumuloMetadataManager#getDefault}
+     *
+     * @return Configured AccumuloMetadataManager class name
+     */
+    @NotNull
+    public String getMetadataManagerClass()
+    {
+        return metaManClass == null || metaManClass.equals("default")
+                ? AccumuloMetadataManager.getDefault(this).getClass().getCanonicalName()
+                : metaManClass;
+    }
+
+    /**
+     * Sets the AccumulMetadataManager class
+     *
+     * @param mmClass Class name of metadata manager, or default
+     */
+    @Config(METADATA_MANAGER_CLASS)
+    public void setMetadataManagerClass(String mmClass)
+    {
+        this.metaManClass = mmClass;
+    }
+
+    /**
+     * Gets the size of the index cardinality cache. Default 100000.
+     *
+     * @return Configured cardinality cache, or 100000 if not set
+     */
+    @NotNull
+    public int getCardinalityCacheSize()
+    {
+        return cardinalityCacheSize == null ? 100000 : cardinalityCacheSize;
+    }
+
+    /**
+     * Sets the cardinality cache size
+     *
+     * @param cardinalityCacheSize Size of the cache
+     */
+    @Config(CARDINALITY_CACHE_SIZE)
+    public void setCardinalityCacheSize(int cardinalityCacheSize)
+    {
+        this.cardinalityCacheSize = cardinalityCacheSize;
+    }
+
+    /**
+     * Gets the expiration, in seconds, of the cardinality cache. Default 300 aka five minutes.
+     *
+     * @return Configured cardinality cache expiration, or 300 if not set
+     */
+    @NotNull
+    public int getCardinalityCacheExpireSeconds()
+    {
+        // 5 minute default
+        return cardinalityCacheExpireSeconds == null ? 300 : cardinalityCacheExpireSeconds;
+    }
+
+    /**
+     * Sets the cardinality cache expiration
+     *
+     * @param cardinalityCacheExpireSeconds Cache expiration value
+     */
+    @Config(CARDINALITY_CACHE_EXPIRE_SECONDS)
+    public void setCardinalityCacheExpireSeconds(int cardinalityCacheExpireSeconds)
+    {
+        this.cardinalityCacheExpireSeconds = cardinalityCacheExpireSeconds;
+    }
+
+    /**
+     * Gets a Boolean value indiciating whether or not the connector should use MiniAccumuloCluster.  This is for testing only.
+     *
+     * @return True if MAC should be used, false otherwise
+     */
+    @NotNull
+    public boolean isMiniAccumuloCluster()
+    {
+        return isMiniAccumuloCluster == null ? false : isMiniAccumuloCluster;
+    }
+
+    /**
+     * Sets whether or not to use MiniAccumuloCluster.  This is for testing only.
+     *
+     * @param isMiniAccumuloCluster True to MAC, false otherwise
+     */
+    @Config(MINI_ACCUMULO_CLUSTER)
+    public void setMiniAccumuloCluster(boolean isMiniAccumuloCluster)
+    {
+        this.isMiniAccumuloCluster = isMiniAccumuloCluster;
+    }
+
+    public static AccumuloConfig fromFile(File f)
```

Comment:
This function is used by some [supporting projects](https://github.com/bloomberg/presto-accumulo) that are not part of this pull request.  These tools should also be part of presto.  It's been suggested to me that they should be transfered to the prestodb org.  I need to follow up on that.